src/promise.js

```diff
@@ -8,19 +8,45 @@ Code distributed by Google as part of the polymer project is also
 subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
 */
 'use strict';
-import ES6Promise from '../node_modules/es6-promise/lib/es6-promise/promise.js';
+import PromisePolyfill from '../node_modules/promise-polyfill/src/index.js';
 
 /*
 Assign the ES6 promise polyfill to window ourselves instead of using the "auto" polyfill
 to work around https://github.com/webcomponents/webcomponentsjs/issues/837
 */
 if (!window.Promise) {
-  window.Promise = ES6Promise;
+  window.Promise = PromisePolyfill;
   // save Promise API
-  ES6Promise.prototype['catch'] = ES6Promise.prototype.catch;
-  ES6Promise.prototype['then'] = ES6Promise.prototype.then;
-  ES6Promise['all'] = ES6Promise.all;
-  ES6Promise['race'] = ES6Promise.race;
-  ES6Promise['resolve'] = ES6Promise.resolve;
-  ES6Promise['reject'] = ES6Promise.reject;
+  /* eslint-disable no-self-assign */
+  // PromisePolyfill.prototype['catch'] = PromisePolyfill.prototype.catch;
```

Comment:
why are these lines commented out?

Reply:
I think those are leftover in v2 from figuring out which properties are renamed.

Reply:
Figured out only `race` and `all` need to be saved with stringification